### PR TITLE
update openai, pydantic, and chromadb versions and examples to work with uniflow

### PR DESCRIPTION
**Pull request by jojortz**
- Update openai version to 1.2.4 to match uniflow
  - change to use OpenAI client.chat.completions instead of ChatCompletion [reference](https://github.com/openai/openai-python/discussions/742)
- Update pydantic to 2.5.2 to match uniflow
  - required chromadb update as well to get around BaseSettings error
- Tested and updated notebooks in these example folders
  - chatbot
  - comparator
  - data_generation
  - retrieval_qa
  - rlhf

### CODE REVIEW

**Comment by jojortz** [comment]
Further updated datasets, transformers, peft, and trl

- For datasets, updated to address this [issue](https://stackoverflow.com/questions/77433096/notimplementederror-loading-a-dataset-cached-in-a-localfilesystem-is-not-suppor)
  - Datasets upgrade required transformers upgrade to avoid conflicts
- With updated torch, got `use_reentrant` ValueError [ref](https://github.com/huggingface/transformers/issues/26969)
  - added `gradient_checkpointing_kwargs` to config and TrainingArguments 
  - updated peft and trl
- With updated trl, got CUDA out of memory, so updated `per_device_train_batch_size` to default to 1. Above 1, get out of memory error on 24GB GPU

File: pykoi/rlhf/config.py [hunk 1]
```diff
@@ -68,7 +66,7 @@ class RLHFConfig:
     #     default=8,
     #     metadata={"help": "Batch size."})
     per_device_train_batch_size: Optional[int] = field(
-        default=2, metadata={"help": "Batch size per device for training."}
+        default=1, metadata={"help": "Batch size per device for training."}
```
Comment (CambioML) [inline pykoi/rlhf/config.py:69]: nit: could you please make a TODO here that for trl 0.7.4 there is a OOM issue and we need to come back to revisit.

**Review by CambioML — COMMENTED** [review]
Also, can you rerun isort and black for the entire pykoi folder.

**Review by CambioML — APPROVED** [review]

File: pykoi/rlhf/config.py (hunk 1 above)
Comment (jojortz) [reply pykoi/rlhf/config.py:70]: added TODO

**Pull request merged**